src/test/java/bio/terra/common/ParquetUtils.java

```diff
@@ -61,12 +64,31 @@
         final Group r = record;
         results.add(
             IntStream.range(0, r.getType().getFields().size())
-                .mapToObj(i -> Pair.of(r.getType().getFieldName(i), r.getValueToString(i, 0)))
+                .mapToObj(
+                    i -> {
+                      if (r.getType()
+                          .getType(i)
+                          .getLogicalTypeAnnotation()
+                          .equals(LogicalTypeAnnotation.uuidType())) {
+                        return Pair.of(
+                            r.getType().getFieldName(i),
+                            getUUIDFromByteArray(r.getBinary(i, 0).getBytes()));
+                      }
+                      return Pair.of(r.getType().getFieldName(i), r.getValueToString(i, 0));
+                    })
                 .collect(Collectors.toMap(Pair::getKey, Pair::getValue)));
       }
     } catch (IOException ex) {
       throw new RuntimeException("Error reading parquet data data", ex);
     }
     return results;
   }
+
+  public static String getUUIDFromByteArray(byte[] bytes) {
```

Comment:
oooooh. I see.  I led you astray.  Great catch!

Reply:
It was the right direction, though!